slime/templates/deployment.yaml

```diff
@@ -0,0 +1,96 @@
+{{- if .Values.deployment.enabled -}}
+{{- $root := . }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "slime.fullname" $root }}
+{{- with .Values.deployment.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  labels:
+    app: {{ template "slime.name" $root }}
+    chart: {{ template "slime.chart" $root }}
+    release: {{ $root.Release.Name }}
+    heritage: {{ $root.Release.Service }}
+{{- with .Values.deployment.labels }}
+{{ toYaml .| indent 4 }}
+{{- end }}
+spec:
+  replicas: {{ .Values.deployment.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "slime.name" $root }}
+      release: {{ $root.Release.Name | quote }}
+{{- with .Values.deployment.strategy }}
+  strategy:
+{{ toYaml . | indent 4 }}
+{{- end }}
+  revisionHistoryLimit: {{ .Values.deployment.revisionHistoryLimit }}
+  template:
+    metadata:
+      name: {{ include "slime.fullname" $root }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+{{- with .Values.deployment.pod.annotations }}
+{{ toYaml . | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "slime.name" $root }}
+        release: {{ $root.Release.Name | quote }}
+{{- with .Values.deployment.pod.labels }}
+{{ toYaml .| indent 8 }}
+{{- end }}
+    spec:
+{{- with .Values.deployment.pod.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+{{- end }}
+      containers:
```

Comment:
You should be able to define multiple containers in a slime chart.

```
containers:
{{ tpl .Values.deployment.spec.template.spec.container $root | indent 6 }}
```
Is there any reason for not doing this?

Reply:
There is no particular reason, just copying the monochart..🙃
Certainly, this way your suggestion may be more flexible!